contracts/contracts/Subcall.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.0;
+
+import {StakingAddress, StakingSecretKey} from "./ConsensusUtils.sol";
+
+/**
+ * @title SDK Subcall wrappers
+ * @dev Interact with Oasis Runtime SDK modules from Sapphire
+ */
+library Subcall {
+    string private constant CONSENSUS_DELEGATE = "consensus.Delegate";
+    string private constant CONSENSUS_UNDELEGATE = "consensus.Undelegate";
+    string private constant CONSENSUS_WITHDRAW = "consensus.Withdraw";
+    string private constant ACCOUNTS_TRANSFER = "accounts.Transfer";
+
+    /// Address of the SUBCALL precompile
+    address internal constant SUBCALL =
+        0x0100000000000000000000000000000000000103;
+
+    /// Only raised if the underlying subcall precompile does not succeed
+    error Subcall_Error();
+
+    /**
+     * Submit a native message to the Oasis runtime layer
+     *
+     * Messages which re-enter the EVM module are forbidden: evm.*
+     *
+     * @param method Native message type
+     * @param body CBOR encoded body
+     * @return status Result of call
+     * @return data CBOR encoded result
+     */
+    function subcall(string memory method, bytes memory body)
+        internal
+        returns (uint64 status, bytes memory data)
+    {
+        (bool success, bytes memory tmp) = SUBCALL.call(
+            abi.encode(method, body)
+        );
+
+        if (!success) {
+            revert Subcall_Error();
+        }
+
+        (status, data) = abi.decode(tmp, (uint64, bytes));
+    }
+
+    /**
+     * @dev Generic method to call `{to:address, amount:uint128}`
+     * @param method Runtime SDK method name ('module.Action')
+     * @param to Destination address
+     * @param value Amount specified
+     * @return status Non-zero on error
+     * @return data Module name on error
+     */
+    function _subcallWithToAndAmount(
+        string memory method,
+        StakingAddress to,
+        uint128 value
+    ) internal returns (uint64 status, bytes memory data) {
+        (status, data) = subcall(
+            method,
+            abi.encodePacked(
+                hex"a262",
+                "to",
+                hex"55",
+                to,
+                hex"66",
+                "amount",
+                hex"8250",
+                value,
+                hex"40"
+            )
+        );
+    }
+
+    error ConsensusUndelegateError(uint64 status, string data);
```

Comment:
Let's move all contract-level errors above after the constants and store them in one place.